src/kalshi.rs

```diff
@@ -82,8 +82,12 @@ pub fn get_mirror_candidates(client: &Client, config: &Settings) -> Result<Vec<K
     info!("{} events listed via Kalshi API", events.len());
     let markets = events
         .into_iter()
-        .map(|event| (&event).try_into())
-        .filter_map(Result::ok)
+        .filter(|e| !e.is_multimarket())
+        .map(|event| {
+            (&event)
+                .try_into()
+                .expect("Failed to convert Kalshi event to market")
```

Comment:
Using [filter_map](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.filter_map) with `.try_into().ok()` would also be an option here.

If you do this the next point will be moot again, but the [suggested message style](https://doc.rust-lang.org/std/error/index.html#common-message-styles) for `expect` is a phrase containing "should", indicating the assumption that was violated.